### PR TITLE
Complete Android SafetyNet attestation

### LINKED ISSUE #48: Android SafetyNet not working
I tested on production the `android-safetynet` attestation. However, it is not working as currently it is not doing any of the checks required by WebAuthn (https://www.w3.org/TR/webauthn/#android-safetynet-attestation) nor the `androidSafetyNetValidateFn()` is returning `true`. 

I also detected this function also needs to include the changes in the journal for it work.

I'm making a PR with this tested changes here and in @JamesCullum [fork](https://github.com/FIDO-Tools/fido2-library). The changes woul be, namely:
- return true if no errors
- perform `nonce` and payload checks
- add the verified checks to journal

### PR DESCRIPTION
**Pull request by martinord**
This PR fixes webauthn-open-source/fido2-lib#48 .
It makes `android-safetynet` attestation format work.

This is the same PR than webauthn-open-source/fido2-lib#49 .

@JamesCullum can you have a look to this PR? If everything is alright, I would appreciate very much a merge and the inclussion in the npm package. I'm using this for my project's dissertation so I did the testing in production already.

I also tagged some commentas as TODO.

Thanks!

### CODE REVIEW

**Comment by JamesCullum** [comment]
The [check is failing due to formatting](https://travis-ci.com/github/FIDO-Tools/fido2-library/jobs/343173265) - can you fix it?

**Comment by martinord** [comment]
Done!

**Comment by JamesCullum** [comment]
Would it be possible to create a test for the new code?
This way it can be better maintained in the future and it doesn't decrease the test coverage 👍

**Comment by martinord** [comment]
@JamesCullum I should say I'm quite a newbie with this type of testing. Never used nyc mocha before and I'm quite lost.

I've been trying to write tests for the validation part I wrote, looking to the tests already written for other attestation formats. However, I don't see how the validation part is being tested. in their correspondent tests.

I assume I have to write it in `test/parseAndroidSafetyNetAttestationDataTest.js`, like the other attestation formats. Can you give me a bit of insight about this? Which tests could I write for the code I wrote?

Thanks for the help

**Comment by JamesCullum** [comment]
So you wrote the code for the function `androidSafetyNetValidateFn` but it is not called in an existing test. This means that someone could break it in the future and we wouldn't notice. I would recommend to use a test for the other function in the file, `androidSafetyNetParseFn`, and use it as foundation to add a test for this method.

Boiled down, a test means "Give it realistic input and verify that it works". If you provide me with all input to this function (JWT etc) I can try to support you with it. You mentioned you were testing it in production already - basically we need to simulate such a test with hardcoded input and check the output.

**Comment by martinord** [comment]
Thanks! The thing is I was not finding were these were tested. The validation of the attestation formats is at `mainTest.js`.

I also tried to use the dataset of `fido2-helpers` like some other tests but the android safetynet was not properly encoded. At the end I added my hardcoded real data to test. 

Hope now everything is alright @JamesCullum ;).

**Comment by JamesCullum** [comment]
Awesome, that's exactly what I meant! Now the formatting follows the same format everywhere else and not only does it offer more functionality, it is also better maintainable in the future. And the coverage even increased, so the numbers look better as well!

Thanks a lot for your great contribution! If there is anything I can do to support with your dissertation, let me know (via email) 👍

**Comment by martinord** [comment]
@JamesCullum many thanks! Let me know if you deploy these changes to `npm` so I can change the dependency in my dissertation.

File: test/mainTest.js [hunk 1]
```diff
@@ -475,6 +476,45 @@ describe("Fido2Lib", function() {
...
+						"attestationObject"
+					),
+					clientDataJSON: coerceToArrayBuffer("eyJ0eXBlIjoid2ViYXV0aG4uY3JlYXRlIiwiY2hhbGxlbmdlIjoiTnJSemdSaEd5NVkwTmxLTmhFQXFzNFpGVmdOR3RONDlaeUNUT2ZMazhHMUVQWTN2bk4zemFzSVp5bmxDQXlVT0xkQjMtQUFMZnkxWEcyTWlWcHNfVnciLCJvcmlnaW4iOiJodHRwczpcL1wvY29udHViZXJuaW8udGljLnVkYy5lcyIsImFuZHJvaWRQYWNrYWdlTmFtZSI6ImNvbS5hbmRyb2lkLmNocm9tZSJ9","clientDataJSON"),
+				},
+			};
+			
+			return serv.attestationResult(makeCredentialAttestationSafetyNetResponse, expectations).then((res) => {
+				console.log(res);
```
Comment (JamesCullum) [inline test/mainTest.js:512]: Logging shouldnt be in the final parts, but as it just runs as part of the test I'll remove it

Comment (martinord) [reply test/mainTest.js:512]: Perfect, forgot to remove it actually

**Comment by JamesCullum** [comment]
I just bumped the version to 2.5.0 and travis automatically deploys the changes in master to npm. So the new version is already available on npm and you can go ahead

**Pull request merged**